### PR TITLE
Target JDK 17 bytecode & APIs

### PR DESCRIPTION
**Pull request by schlosna**
## Before this PR

https://github.com/palantir/tritium/pull/1956 is failing due to `delegate-processors:1.2.0` bump targeting JDK 17 bytecode via https://github.com/palantir/delegate-processors/pull/105

## After this PR

==COMMIT_MSG==
Target JDK 17 bytecode & APIs
==COMMIT_MSG==

### CODE REVIEW

**Review by pkoenig10 — DISMISSED** [review]

**Comment by schlosna** [comment]
Holding off merging pending confirmation of any JDK 11 stragglers

File: build.gradle [hunk 1]
```diff
@@ -31,7 +31,7 @@ apply plugin: 'com.palantir.baseline-java-versions'
 apply plugin: 'com.palantir.jdks.latest'
 
 javaVersions {
-    libraryTarget = 11
+    libraryTarget = 17
```
Comment (carterkozak) [inline build.gradle]: This piece has been excavated: #2038

File: .java-version
Comment (bmarcaur) [inline .java-version:1]: This is no longer necessary with https://github.com/palantir/tritium/pull/2013

**Review by bmarcaur — DISMISSED** [review]

**Review by bmarcaur — APPROVED** [review]

**Comment by schlosna** [comment]
unleash the hounds

**Review by carterkozak — APPROVED** [review]

**Pull request merged**